schemas/pioneer-citp-news-disinfo-two/rallymeasurements/rallymeasurements.2.schema.json

```diff
@@ -0,0 +1,265 @@
+{
+  "$comment": "Information about this study is available at https://github.com/citp/news-disinformation-study",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "mozPipelineMetadata": {
+    "bq_dataset_family": "pioneer_citp_news_disinfo_two",
+    "bq_metadata_format": "pioneer",
+    "bq_table": "rallymeasurements_v2",
+    "expiration_policy": {
+      "delete_after_days": 180
+    }
+  },
+  "properties": {
+    "newsAndDisinfo.linkExposure": {
+      "additionalProperties": false,
```

Comment:
We do try to get our extension updates and schema updates out close to simultaneously. Iirc, the top level of the schema *does* allow additional properties (that's actually what's happening right now -- the extension update got through review and out to users really quickly, so the new schema fields are being dumped into the top-level addtl properties column).

For this study, schema updates are somewhat rare -- I think I've made no more than 3-4 in the last two years or so. This could plausibly be the last one. I think in all of the other cases, we did get the timing right, and the schema was live before the new data was flowing.

I think that disallowing additional properties throughout most the schema is "right" in the sense that it matches the ethos of the study: if a freak bug caused an extra field to be added to a ping, that data truly shouldn't be available to us. Allowing additional properties at the top level is helpful in cases like the current one, where we've added a new section and the two updates are just misaligned by a few days. (Given how the analysis code is written, I struggle to see how a bug could add a top-level field, but a field within one of the sections isn't *completely* implausible).

All that to say that I don't see a clear right answer here, but I think it's likely that the current setup is reasonable.

Reply:
Great, sounds good. Thanks for the thoughtful response, and context!